src/plot_api/plot_api.js

```diff
@@ -3782,8 +3782,11 @@ function makePlotFramework(gd) {
         .classed('shapelayer', true);
 
     // single cartesian layer for the whole plot
-    fullLayout._cartesianlayer = fullLayout._paper.append('g').classed('cartesianlayer', true);
-
+    fullLayout._cartesianlayerBelow = fullLayout._paper.append('g').classed('cartesianlayer-below', true);
+    var layerBetween = fullLayout._paper.append('g').classed('layer-between', true);
+    fullLayout.imageLayerBetween = layerBetween.append('g').classed('imagelayer', true);
```

Comment:
We shouldn't add new attribute to `fullLayout` that are not part of the API unless the key is underscored e.g. `_imageLayerBetween` is acceptable.